opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/AnalyzerFramework.java

```diff
@@ -0,0 +1,400 @@
+import org.opengrok.indexer.analysis.data.IgnorantAnalyzerFactory;
+import org.opengrok.indexer.analysis.data.ImageAnalyzerFactory;
+import org.opengrok.indexer.analysis.document.MandocAnalyzerFactory;
+import org.opengrok.indexer.analysis.document.TroffAnalyzerFactory;
+import org.opengrok.indexer.analysis.eiffel.EiffelAnalyzerFactory;
+import org.opengrok.indexer.analysis.erlang.ErlangAnalyzerFactory;
+import org.opengrok.indexer.analysis.executables.ELFAnalyzerFactory;
+import org.opengrok.indexer.analysis.executables.JarAnalyzerFactory;
+import org.opengrok.indexer.analysis.executables.JavaClassAnalyzerFactory;
+import org.opengrok.indexer.analysis.fortran.FortranAnalyzerFactory;
+import org.opengrok.indexer.analysis.golang.GolangAnalyzerFactory;
+import org.opengrok.indexer.analysis.haskell.HaskellAnalyzerFactory;
+import org.opengrok.indexer.analysis.java.JavaAnalyzerFactory;
+import org.opengrok.indexer.analysis.javascript.JavaScriptAnalyzerFactory;
+import org.opengrok.indexer.analysis.json.JsonAnalyzerFactory;
+import org.opengrok.indexer.analysis.kotlin.KotlinAnalyzerFactory;
+import org.opengrok.indexer.analysis.lisp.LispAnalyzerFactory;
+import org.opengrok.indexer.analysis.lua.LuaAnalyzerFactory;
+import org.opengrok.indexer.analysis.pascal.PascalAnalyzerFactory;
+import org.opengrok.indexer.analysis.perl.PerlAnalyzerFactory;
+import org.opengrok.indexer.analysis.php.PhpAnalyzerFactory;
+import org.opengrok.indexer.analysis.plain.PlainAnalyzerFactory;
+import org.opengrok.indexer.analysis.plain.XMLAnalyzerFactory;
+import org.opengrok.indexer.analysis.powershell.PowershellAnalyzerFactory;
+import org.opengrok.indexer.analysis.python.PythonAnalyzerFactory;
+import org.opengrok.indexer.analysis.ruby.RubyAnalyzerFactory;
+import org.opengrok.indexer.analysis.rust.RustAnalyzerFactory;
+import org.opengrok.indexer.analysis.scala.ScalaAnalyzerFactory;
+import org.opengrok.indexer.analysis.sh.ShAnalyzerFactory;
+import org.opengrok.indexer.analysis.sql.PLSQLAnalyzerFactory;
+import org.opengrok.indexer.analysis.sql.SQLAnalyzerFactory;
+import org.opengrok.indexer.analysis.swift.SwiftAnalyzerFactory;
+import org.opengrok.indexer.analysis.tcl.TclAnalyzerFactory;
+import org.opengrok.indexer.analysis.uue.UuencodeAnalyzerFactory;
+import org.opengrok.indexer.analysis.vb.VBAnalyzerFactory;
+import org.opengrok.indexer.framework.PluginFramework;
+import org.opengrok.indexer.logger.LoggerFactory;
+
+/**
+ * A framework for loading the analyzers from .jar or .class files.
+ */
+public class AnalyzerFramework extends PluginFramework<IAnalyzerPlugin, AnalyzersInfo> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AnalyzerFramework.class);
+
+    /**
+     * The default {@code FileAnalyzerFactory} instance.
+     */
+    public static final AnalyzerFactory DEFAULT_ANALYZER_FACTORY = new FileAnalyzerFactory();
+
+    private static final IAnalyzerPlugin[] DEFAULT_PLUGINS = {
```

Comment:
I assume this is a temporary measure until the framework is fully implemented and in the final state it will dynamically load all the plugins from a directory - no hard-coding needed.

Reply:
Yes this is for seamless transition from the old approach to the new approach.

That is a good point though. I'm affraid that with that approach we would need to distribute two jar files instead of one and more to force people to actually put the second one somewhere (plugins directory) and point the configuration directory there and I think this is a little bit more than what I'd expect to get opengrok working.